Account/FxAClient10.swift

```diff
@@ -172,44 +169,68 @@ public class FxAClient10 {
     }
 
     private class func keysResponseFromJSON(keyRequestKey: NSData, json: JSON) -> FxAKeysResponse? {
-        if json.isError {
+        guard !json.isError,
+            let bundle = json["bundle"].asString else {
+                return nil
+        }
+
+        let data = bundle.hexDecodedData
+        guard data.length == 3 * KeyLength else {
+            return nil
+        }
+
+        let ciphertext = data.subdataWithRange(NSMakeRange(0 * KeyLength, 2 * KeyLength))
+        let MAC = data.subdataWithRange(NSMakeRange(2 * KeyLength, 1 * KeyLength))
+
+        let salt: NSData = NSData()
+        let contextInfo: NSData = KW("account/keys")
+        let bytes = keyRequestKey.deriveHKDFSHA256KeyWithSalt(salt, contextInfo: contextInfo, length: UInt(3 * KeyLength))
+        let respHMACKey = bytes.subdataWithRange(NSMakeRange(0 * KeyLength, 1 * KeyLength))
+        let respXORKey = bytes.subdataWithRange(NSMakeRange(1 * KeyLength, 2 * KeyLength))
+
+        guard ciphertext.hmacSha256WithKey(respHMACKey) == MAC else {
+            NSLog("Bad HMAC in /keys response!")
```

Comment:
Wish we could do more than log here, but this probably means the server is buggy rather than that the Sync data is corrupt.